react/I18n/index.jsx

```diff
@@ -94,7 +95,22 @@ export const translate = () => WrappedComponent => {
   return Wrapper
 }
 
-export const useI18n = () => useContext(I18nContext)
+export const useI18n = () => {
+  return useContext(I18nContext)
+}
+
+export const createUseI18n = locales => () => {
+  const { lang } = useI18n() || { lang: DEFAULT_LANG }
+  return useMemo(() => {
+    const polyglot = new Polyglot({
+      lang: lang,
+      phrases: locales[lang]
+    })
+    const f = initFormat(lang)
+    const t = polyglot.t.bind(polyglot)
+    return { t, f, lang }
+  }, [lang])
+}
```

Comment:
Awesome! Is it possible to add a test for this method to test it and also to have an example of how to use it?